pkg/controller/v1alpha2/core/scopes/healthscope/healthscope.go

```diff
@@ -151,24 +156,36 @@ func CheckDeploymentHealth(ctx context.Context, client client.Client, ref runtim
 		r.Diagnosis = errors.Wrap(err, errHealthCheck).Error()
 		return r
 	}
-	r.Target.UID = deployment.GetUID()
+	compName, exist := deployment.GetLabels()[oam.LabelAppComponent]
+	if !exist {
+		compName = "UNKNOWN NAME"
+	}
+	r.ComponentName = compName
```

Comment:
I see this code twice, can we make it a common function, just put the `runtime.Object` as argument